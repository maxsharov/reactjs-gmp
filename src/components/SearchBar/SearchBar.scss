.search-bar {
  padding: 0 120px;

  h2 {
    font-size: calc(var(--page-pad) * 2.5);
    font-weight: 300;
    text-transform: uppercase;
    color: #FFFFFF;
  }

  &--content {
    display: flex;

    .search {
      margin-right: 14px;
      width: 100%;
      height: 57px;
    }

    .search-button {
      width: 233px;
    }
  }
}

