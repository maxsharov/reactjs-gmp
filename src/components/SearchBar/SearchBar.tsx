import React from 'react'

import Button from "../UI/Button"

import styles from './SearchBar.scss'

const SearchBar = () => {
  return (
    <div className={styles["search-bar"]}>
      <h2>Find your movie</h2>
      <div className={styles["search-bar--content"]}>
        <input type="text" />
        <Button
          primary
          classes={styles["search-bar-btn"]}
        >
          Search
        </Button>
      </div>
    </div>
  )
}

export default SearchBar