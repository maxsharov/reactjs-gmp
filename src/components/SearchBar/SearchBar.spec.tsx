import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import SearchBar from './SearchBar'

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('renders correctly', () => {
  const tree = renderer
    .create(<SearchBar />)
    .toJSON();

  expect(tree).toMatchSnapshot();
})

describe('SearchBar', () => {
  it('should show search input and button', () => {
    const { container } = render(<SearchBar />)

    expect(container.querySelector('.search')).toBeInTheDocument()
    expect(container.querySelector('.search-button')).toBeInTheDocument()
  })

  it('should run handle on button click', () => {
    const { getByText } = render(<SearchBar />)

    const button = getByText(/search/i)

    fireEvent.click(button)

    expect(mockedUsedNavigate).toBeCalledTimes(1)
  })
})

