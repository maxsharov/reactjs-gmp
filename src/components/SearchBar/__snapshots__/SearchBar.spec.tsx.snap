// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search-bar"
>
  <h2>
    Find your movie
  </h2>
  <div
    className="search-bar--content"
  >
    <input
      className="search"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="button button--primary search-button"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </div>
</div>
`;
